src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -702,83 +694,84 @@ public BuildingTownHall getTownHall()
     }
 
     @Override
-    public boolean hasTownHall() { return townHall != null; }
+    public boolean hasTownHall()
+    {
+        return townHall != null;
+    }
 
     /**
      * Get building in Colony by ID
      *
      * @param buildingId    ID (coordinates) of the building to get
-     * @return              Building belonging to the given ID
+     * @return              AbstractBuilding belonging to the given ID
      */
-    public Building getBuilding(BlockPos buildingId)
+    public AbstractBuilding getBuilding(BlockPos buildingId)
     {
         return buildings.get(buildingId);
     }
 
-    //TODO document
     /**
-     * Get building in Colony by ID
+     * Get building in Colony by ID. The building will be casted to the provided type.
      *
      * @param buildingId    ID (coordinates) of the building to get
      * @param type          Type of building
-     * @param <BUILDING>
-     * @return
+     * @param <B>           Building class
+     * @return the building with the specified id.
      */
-    public <BUILDING extends Building> BUILDING getBuilding(BlockPos buildingId, Class<BUILDING> type)
+    public <B extends AbstractBuilding> B getBuilding(BlockPos buildingId, Class<B> type)
     {
         try
         {
             return type.cast(buildings.get(buildingId));
         }
-        catch (ClassCastException ignored) {}
-
-        return null;
+        catch (ClassCastException e)
+        {
+            Log.logger.warn("getBuilding called with wrong type: ", e);
+            return null;
+        }
     }
 
     /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)